core/services/workflows/metering/metering.go

```
@@ -180,10 +181,17 @@ func (r *Report) Reserve(ctx context.Context) error {
 
 // ConvertToBalance converts a resource dimensions amount to a credit amount.
 func (r *Report) ConvertToBalance(fromUnit string, amount decimal.Decimal) (decimal.Decimal, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
 	if !r.ready {
 		return decimal.Zero, ErrNoReserve
 	}
 
+	if r.meteringMode {
```

would a consumer be confused this? should we return a metering mode err so that we don't end up setting the `Deduct` field on the metering report?

---

I wasn't certain on this. We probably should return an error and maybe the same error as the case of switching to metering mode. I'll add this.

---

My vote is for same behavior in both scenarios